Core/src/org/sleuthkit/autopsy/mainui/nodes/actions/DeleteAnalysisResultSetAction.java

```diff
@@ -0,0 +1,115 @@
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Action class for Deleting Analysis Result objects.
+ */
+public class DeleteAnalysisResultSetAction extends AbstractAction {
+    
+    @Messages({
+        "DeleteAnalysisResultsAction.label=Delete Analysis Results",
+        "DeleteAnalysisResultsAction.title=Deleting Analysis Results",
+        "# {0} - result type", 
+        "DeleteAnalysisResultsAction.progress.allResults=Deleting Analysis Results type {0}",
+        "# {0} - result type", "# {1} - configuration", 
+        "DeleteAnalysisResultsAction.progress.allResultsWithConfiguration=Deleting Analysis Results type {0} and configuration {1}",
+        "# {0} - result type", 
+        "DeleteAnalysisResultsAction.warning.allResults=Are you sure you want to delete all Analysis Results of type {0}?",
+        "# {0} - result type", "# {1} - configuration", 
+        "DeleteAnalysisResultsAction.warning.allResultsWithConfiguration=Are you sure you want to delete all Analysis Results of type {0} and configuration {1}?"        
+    })
+    
+    private static final Logger logger = Logger.getLogger(DeleteAnalysisResultSetAction.class.getName());
+    private static final long serialVersionUID = 1L;
+    
+    private final BlackboardArtifact.Type type;
+    private final String configuration;
+    private final Long dsID;
+    
+    public DeleteAnalysisResultSetAction(BlackboardArtifact.Type type, String configuration, Long dsID) {
+        super(Bundle.DeleteAnalysisResultsAction_label());
+        this.type = type;
+        this.configuration = configuration;
+        this.dsID = dsID;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        SwingWorker<Void, Void> worker = new SwingWorker<Void, Void>() {
+            @Override
+            protected Void doInBackground() throws Exception {
+
+                AppFrameProgressBar progress = new AppFrameProgressBar(Bundle.DeleteAnalysisResultsAction_title());
+                try {
+                    String progressMessage;
+                    String warningMessage;
+                    if (configuration == null || configuration.isEmpty()) {
+                        progressMessage = Bundle.DeleteAnalysisResultsAction_progress_allResults(type.getDisplayName());
+                        warningMessage = Bundle.DeleteAnalysisResultsAction_warning_allResults(type.getDisplayName());
+                    } else {
+                        progressMessage = Bundle.DeleteAnalysisResultsAction_progress_allResultsWithConfiguration(type.getDisplayName(), configuration);
+                        warningMessage = Bundle.DeleteAnalysisResultsAction_warning_allResultsWithConfiguration(type.getDisplayName(), configuration);
+                    }
+ 
+                    int response = JOptionPane.showConfirmDialog(
+                            WindowManager.getDefault().getMainWindow(),
+                            warningMessage,
```

Comment:
@eugene7646 Swing is not thread safe. You cannot do gui stuff in a background thread.

You will need to use SwingUtilities.invokeAndWait for the confirmation dialog.

Reply:
These calls might need to be in the EDT too.